### PR TITLE
patchelf@0.18.0 breaks intel compiler libraries

### PR DESCRIPTION
**Pull request by stephenmsachs**

### CODE REVIEW

**Comment by stephenmsachs** [comment]
Fixes https://github.com/spack/spack/issues/39252

File: var/spack/repos/builtin/packages/intel-oneapi-compilers/package.py [hunk 1]
```diff
@@ -166,6 +166,8 @@ class IntelOneapiCompilers(IntelOneApiPackage):
     homepage = "https://software.intel.com/content/www/us/en/develop/tools/oneapi.html"
 
     depends_on("patchelf", type="build")
+    # See https://github.com/spack/spack/issues/39252
+    conflicts("patchelf@0.18.0")
```
Comment (haampie) [inline var/spack/repos/builtin/packages/intel-oneapi-compilers/package.py]:
```suggestion
    conflicts("patchelf@0.18:")
```

**Review by haampie — DISMISSED** [review]

File: var/spack/repos/builtin/packages/intel-oneapi-compilers/package.py [hunk 2]
```diff
@@ -166,6 +166,8 @@ class IntelOneapiCompilers(IntelOneApiPackage):
     homepage = "https://software.intel.com/content/www/us/en/develop/tools/oneapi.html"
 
     depends_on("patchelf", type="build")
+    # See https://github.com/spack/spack/issues/39252
+    conflicts("patchelf@0.18:")
```
Comment (haampie) [inline var/spack/repos/builtin/packages/intel-oneapi-compilers/package.py]:
```suggestion
    # See https://github.com/spack/spack/issues/39252
    depends_on("patchelf@:0.17", type="build")
```

**Review by haampie — APPROVED** [review]

**Comment by rscohn2** [comment]
Thanks for the fix.

**Pull request merged**